modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/programs/Genesis.scala

```diff
@@ -58,30 +65,36 @@ object Genesis {
         .flatMap(OwnCollateralNotSatisfied.raiseError[F, Unit].unlessA)
 
       signedBinary <- stateChannelSnapshotService.createGenesisBinary(hashedGenesis.signed)
-      signedBinaryHash <- signedBinary.hashF
+      address = signedBinary.value.toAddress
+      _ <-
+        if (identifier =!= address) IdentifierNotEqualAddressFromGenesisDataError(identifier, address).raiseError[F, Unit]
+        else Applicative[F].unit
+      binaryHash <- signedBinary.toHashed.map(_.hash)
```

Comment:
isn't it `signedBinaryHash`?

Reply:
It has been changed to hash just from binary, so the change of name.

Reply:
what is a difference between `signedBinary.hashF` and `signedBinary.toHashed.map(_.hash)`. Isn't it the same hash? if no, then it's fine